### PR TITLE
Generate error when private methods are annotated with @Around advice

### LINKED ISSUE #5522: Warn or error out when AOP annotations are placed on private methods
We had some code that got refactored and a `@Transactional` annotation got placed on a private method accidentally. We only discovered it thanks to some failing tests. It'd be nice if Micronaut's annotation processors refused to compile or at least emitted a warning in such a case. Thanks!

### PR DESCRIPTION
**Pull request by mattmoss**
Fixes #5522

### CODE REVIEW

File: inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java [hunk 1]
```diff
@@ -804,6 +804,10 @@ public Object visitExecutable(ExecutableElement method, Object o) {
                             (isExecutableThroughType(method.getEnclosingElement(), methodAnnotationMetadata, annotationMetadata, modifiers, isPublic) ||
                                     hasAroundStereotype(annotationMetadata));
 
+            if (javaMethodElement.isPrivate() && hasAroundStereotype(annotationMetadata)) {
```
Comment (jameskleeh) [inline inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java]: Perhaps using `isExecutable` here is more complete?

Comment (jameskleeh) [reply inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java]: Oh I see that variable includes `!hasInvalidModifiers`. I would say to remove that condition then check it inside the `if (isDeclaredBean && isExecutable) {` condition. This check also needs done for Groovy

Comment (mattmoss) [reply inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java:807]: I had initially replicated this in Groovy AST code, but private methods don't show up as private.

File: inject-groovy/src/test/groovy/io/micronaut/aop/simple/AnotherClass.groovy [hunk 2]
```diff
@@ -29,11 +29,6 @@ class AnotherClass<A> {
         return "Name is " + name
     }
 
-    // protected methods not proxied
```
Comment (jameskleeh) [inline inject-groovy/src/test/groovy/io/micronaut/aop/simple/AnotherClass.groovy]: Can you revert this? This method should not cause a problem

Comment (mattmoss) [reply inject-groovy/src/test/groovy/io/micronaut/aop/simple/AnotherClass.groovy]: Reverted.

**Comment by dstepanov** [comment]
Is it fixing https://github.com/micronaut-projects/micronaut-data/issues/882 ?

**Comment by jameskleeh** [comment]
@dstepanov Yep

**Pull request merged**